src/View/Helper/DateRangesHelper.php

```diff
@@ -123,4 +123,18 @@ public function getClosestRange(array $ranges, string $now = null): ?DateRange
 
         return $sorted->last(); // Get closest event in the past, relative to `$now`.
     }
+
+    /**
+     * Sort a list of objects by their closest date.
+     *
+     * @param \BEdita\Core\Model\Entity\ObjectEntity[] $objects Objects to sort.
+     * @return \BEdita\Core\Model\Entity\ObjectEntity[] Sorted objects.
+     */
+    public function sortByClosestRange(iterable $objects): array
+    {
+        $it = collection($objects)->toArray();
+        usort($it, fn ($a, $b): int => $this->getClosestRange($a->date_ranges ?? [])->start_date->getTimestamp() - $this->getClosestRange($b->date_ranges ?? [])->start_date->getTimestamp());
+
+        return $it;
```

Comment:
It should be possible to use [`Collection::sortBy()`](https://book.cakephp.org/3/en/core-libraries/collections.html#Cake\Collection\Collection::sortBy) method instead:

```suggestion
        return collection($objects)
            ->sortBy(fn (ObjectEntity $obj): int => $this->getClosestRange($a->date_ranges ?? [])->start_date->getTimestamp());
```

However, I think it should be taken into account what to do when `$this->getClosestRange()` returns `null`, that is when an object has no associated date ranges.